dmutils/filters.py

```diff
@@ -13,3 +15,26 @@ def smartjoin(input):
         return '{}'.format(list_to_join[0])
     else:
         return ''
+
+
+def format_links(text):
+    pattern = re.compile(r"""((?:http[s]?://|www)(?:[a-zA-Z]|[0-9]|[$-_@.&#+]|[!*\(\),]|
+                        (?:%[0-9a-fA-F][0-9a-fA-F]))+)""", re.X)
```

Comment:
This regex seems far too liberal in what it will match.
eg. `www.`, `http://.`, `www.a`

It should probably check for a minimum number of `.` separated things in the domain (at least two), and ideally that there is a valid top-level domain at the end.

Also, once a possible URL has been extracted it would be nice to sanity-check it by reading it in with something like `urlparse`.  That doesn't guarantee it's right, but would give a bit more confidence.